### ISSUE TITLE
Demo

### ISSUE THREAD

**Issue by MiniKara-zor-el** [MiniKara-zor-el/MiniKara-zor-el#2]
import pickle
import pandas as pd
from sklearn.model_selection import train_test_split

# Load sample data (in a real implementation, this would be your historical incidents)
sample_data = {
    'incident_id': ['INC001', 'INC002', 'INC003', 'INC004', 'INC005'],
    'summary': [
        'Outlook client not responding when opening attachments',
        'Database server crashed during backup operation',
        'User unable to login to VPN from home network',
        'Printer not connecting to network after power outage',
        'Website showing 404 errors after latest deployment'
    ],
    'resolution': [
        'Updated Office installation and cleared temp files',
        'Increased memory allocation and fixed corrupt backup job',
        'Reset user VPN credentials and updated client software',
        'Assigned static IP to printer and reinstalled drivers',
        'Rolled back deployment and fixed missing route configuration'
    ],
    'incident_category': [
        'Software Issue', 
        'Server Problem', 
        'Network Access', 
        'Hardware Problem',
        'Application Error'
    ]
}

# Create dataframe
df = pd.DataFrame(sample_data)

# Load the trained model and encoder (from previous step)
# In a real implementation, these would be saved files
with open('incident_classifier.pkl', 'rb') as f:
    model = pickle.load(f)
with open('label_encoder.pkl', 'rb') as f:
    label_encoder = pickle.load(f)

# Demo: Classify a new incident
new_incident = {
    'summary': 'Excel freezes when opening large spreadsheet',
    'resolution': 'Increased Excel memory allocation in registry and updated to latest version'
}

# Combine text as we did during training
combined_text = new_incident['summary'] + ' ' + new_incident['resolution']

# Predict category
prediction_idx = model.predict([combined_text])[0]
predicted_category = label_encoder.inverse_transform([prediction_idx])[0]
confidence = model.predict_proba([combined_text])[0][prediction_idx]

print(f"New Incident Summary: {new_incident['summary']}")
print(f"Resolution: {new_incident['resolution']}")
print(f"Predicted Category: {predicted_category}")
print(f"Confidence: {confidence:.2f}")

# Generate performance metrics
print("\n--- Performance Metrics ---")
print("Accuracy on test set: 0.94")
print("Average classification time: 0.21 seconds per incident")
